packages/suite-web/vite.config.ts

```diff
@@ -0,0 +1,162 @@
+
+import { suiteVersion } from '../suite/package.json';
+
+// Use require instead of import for TypeScript files
+const { assetPrefix, project } = require('../suite-build/utils/env');
+
+// Plugin to serve static files with /static prefix
+const staticAliasPlugin = (): Plugin => ({
+    name: 'static-alias',
+    configureServer(server: ViteDevServer) {
+        // Middleware to handle /static requests
+        server.middlewares.use((req, _res, next) => {
+            if (req.url?.startsWith('/static/')) {
+                // Rewrite the URL to access the file from the public directory
+                req.url = req.url.replace('/static/', '/');
+            }
+            next();
+        });
+    },
+});
+
+// Plugin to serve core.js in dev mode
+
+// Plugin to handle workers similar to webpack's worker-loader
+const workerPlugin = (): Plugin => ({
+    name: 'worker-loader',
+    transform(_code, id) {
+        if (/\/workers\/[^/]+\/index\.ts$/.test(id)) {
+            // Return a virtual module that creates a web worker
+            return {
+                code: `
+                    const worker = () => new Worker(new URL('${id}', import.meta.url), { type: 'module' });
+                    export default worker;
+                `,
+                map: null,
+            };
+        }
+    },
+});
+
+const serveCorePlugin = () => ({
+    name: 'serve-core',
+    configureServer(server: ViteDevServer) {
+        server.middlewares.use(async (req, res, next) => {
+            if (req.url?.endsWith('/js/core.js')) {
+                const code = await server.transformRequest(
+                    resolve(__dirname, '../connect/src/core/index.ts'),
+                    { ssr: false },
+                );
+                if (code?.code) {
+                    res.setHeader('Content-Type', 'application/javascript');
+                    res.end(code.code);
+
+                    return;
+                }
+            }
+            next();
+        });
+    },
+});
+
+// This helper creates aliases for all workspace packages
+const createWorkspaceAliases = () => {
+    const suiteCommonAliases = Object.fromEntries(
+        readdirSync(resolve(__dirname, '../../suite-common'), { withFileTypes: true })
+            .filter(dirent => dirent.isDirectory())
+            .map(dirent => [
+                `@suite-common/${dirent.name}`,
+                resolve(__dirname, '../../suite-common', dirent.name),
+            ]),
+    );
+
+    const trezorPackagesAliases = Object.fromEntries(
+        readdirSync(resolve(__dirname, '../'), { withFileTypes: true })
+            .filter(dirent => dirent.isDirectory() && dirent.name !== 'suite-web')
+            .map(dirent => [`@trezor/${dirent.name}`, resolve(__dirname, '../', dirent.name)]),
+    );
+
+    return Object.entries({
+        ...suiteCommonAliases,
+        ...trezorPackagesAliases,
+    }).map(([find, replacement]) => ({
+        find,
+        replacement,
+    }));
```

Comment:
I meant this with the earlier comment – much simpler to do the same thing :slightly_smiling_face: 
```suggestion
const suiteCommonAliases = readdirSync(resolve(__dirname, '../../suite-common'), {
        withFileTypes: true,
    })
        .filter(dirent => dirent.isDirectory())
        .map(dirent => ({
            find: `@suite-common/${dirent.name}`,
            replacement: resolve(__dirname, '../../suite-common', dirent.name),
        }));

    const trezorPackagesAliases = readdirSync(resolve(__dirname, '../'), { withFileTypes: true })
        .filter(dirent => dirent.isDirectory() && dirent.name !== 'suite-web')
        .map(dirent => ({
            find: `@trezor/${dirent.name}`,
            replacement: resolve(__dirname, '../', dirent.name),
        }));

    return [...suiteCommonAliases, ...trezorPackagesAliases];
```

Reply:
oki